examples/pet-mad-uq/pet-mad-uq.py

```diff
@@ -0,0 +1,430 @@
+the following scaffold. For more information on loading a dataset with the
+infrastructure, have a look at
+`this section <https://metatensor.github.io/metatrain/latest/dev-docs/utils/data>`_
+of the documentation. The pseudocode below also shows how to create an ensemble model
+from the last-layer parameters of a model.
+
+.. code-block:: python
+
+    from metatrain.utils.llpr import LLPRUncertaintyModel
+    from metatomic.torch import AtomisticModel, ModelMetadata
+
+    # You need to provide a model and datasets (wrapped in PyTorch dataloaders).
+    model = ...
+    dataloaders = {"train": ..., "val": ...}
+
+    # Wrap the model in a module capable of estimating uncertainties, estimate the
+    # inverse covariance on the training set, and calibrate the model on the validation
+    # set.
+    llpr_model = LLPRUncertaintyModel(model)
+    llpr_model.compute_covariance(dataloaders["train"])
+    llpr_model.compute_inverse_covariance(regularizer=1e-4)
+    llpr_model.calibrate(dataloaders["val"])
+
+    # In the next step, we show how to enable ensembles in PET-MAD. For that, it is
+    # necessary to extract the last-layer parameters of the model. The ensemble
+    # generation expects the parameters in a flat-vector format.
+    last_layer_parameters = ...
+
+    # Generate an ensemble with 128 members to compare ensemble uncertainties to LLPR
+    # scores.
+    llpr_model.generate_ensemble({"energy": last_layer_parameters}, 128)
+
+    # Save the model to disk using metatomic.
+    exported_model = AtomisticModel(
+        llpr_model.eval(),
+        ModelMetadata(),
+        llpr_model.capabilities,
+    )
+    exported_model.save("models/model-with-llpr.pt")
+
+At the bottom of the page, you'll find a ZIP file containing the whole example. Note
+that it comes with an `environment.yml` file specifying all dependencies required
+to execute the script.
+"""
+
+# %%
+import os
+import subprocess
+from urllib.request import urlretrieve
+
+import ase.cell
+import ase.ga.utilities
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import torch
+from ase import Atoms
+from ase.filters import FrechetCellFilter
+from ase.io.cif import read_cif
+from ase.optimize.bfgs import BFGS
+from ipi.utils.scripting import InteractiveSimulation
+from metatomic.torch import ModelEvaluationOptions, ModelOutput
+from metatomic.torch.ase_calculator import MetatomicCalculator
+from metatrain.utils.data import Dataset, read_systems, read_targets
+from metatrain.utils.data.system_to_ase import system_to_ase
+
+
+# %%
+# Model Preparation
+# -----------------
+# All examples require a PET-MAD model with ensemble and LLPR prediction. The
+# following
+# code loads a pre-built model using the ASE-compatible calculator wrapper. Using the
+# calculator instead of calling the model directly conveniently hides computing
+# neighbor lists in the calculator. By default, PET-MAD does not (yet) come equipped
```

Comment:
By the way, this is not true anymore. We changed the structure to publish a single LLPR-enabled model. There will be no published model without UQ starting from the next version of PET-MAD. Please verify with @abmazitov. We can also wait until that model is available through HuggingFace

Reply:
I removed the last sentence of the section in 89b3bd0 for now. But you're right, this is something that needs to be changed when the UQ-enabled model becomes available on Hugging Face.